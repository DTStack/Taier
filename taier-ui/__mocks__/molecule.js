const { editor, ...rest } = require('./monaco-editor');

module.exports = {
    Component: class {
        emit = jest.fn();
        subscribe = jest.fn();
        getState = () => this.state;
        setState = (state) => Object.assign(this.state, state);
    },
    FolderTreeService: class FolderTreeService {
        get = jest.fn();
    },
    GlobalEvent: class {
        emit = jest.fn();
        subscribe = jest.fn();
    },
    panel: {
        getState: jest.fn(),
        setState: jest.fn(),
        setActive: jest.fn(),
        add: jest.fn(),
        update: jest.fn(),
        getPanel: jest.fn(),
    },
    explorer: {
        forceUpdate: jest.fn(),
    },
    editor: {
        updateActions: jest.fn(),
        updateGroup: jest.fn(),
        updateTab: jest.fn(),
        getDefaultActions: jest.fn(),
        getState: jest.fn(),
        getGroupIdByTab: jest.fn(),
        setActive: jest.fn(),
        isOpened: jest.fn(),
        open: jest.fn(),
        closeTab: jest.fn(),
        ...editor,
    },
    folderTree: {
        get: jest.fn(),
        add: jest.fn(),
        getState: jest.fn(),
        update: jest.fn(),
        setActive: jest.fn(),
    },
    layout: {
        getState: jest.fn(),
        togglePanelVisibility: jest.fn(),
    },
    Header: ({ title, toolbar }) => (
        <div data-testid="mockHeader">
            {title}
            <div data-testid="mockHeaderToolbar">{toolbar}</div>
        </div>
    ),
    Content: ({ children }) => <div data-testid="mockContent">{children}</div>,
    FolderTree: jest.fn(),
    connect: jest.fn((_, children) => children),
    TreeViewUtil: jest.fn(),
    FileTypes: {},
    TreeNodeModel: class {
        constructor(params) {
            Object.assign(this, params);
        }
    },
    colorTheme: {
        getColorThemeMode: jest.fn(),
        onChange: jest.fn(),
    },
    notification: {
        getState: jest.fn(),
        add: jest.fn(),
    },
    component: {
        Scrollbar: ({ children }) => <div data-testid="Scrollbar">{children}</div>,
        Icon: ({ type }) => <svg data-testid="Icon" type={type} />,
        ActionBar: ({ data = [] }) => (
            <div data-testid="ActionBar">
                {data.map((i) => (
                    <div key={i.id} id={i.id}>
                        {i.name}
                    </div>
                ))}
            </div>
        ),
    },
    sidebar: {
        setActive: jest.fn(),
    },
    activityBar: {
        setActive: jest.fn(),
    },
    ...rest,
};
