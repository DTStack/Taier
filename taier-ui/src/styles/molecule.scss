@import "./variables.scss";

.mo-menuBar--horizontal {
    height: $menuBar-size;
    background: var(--activityBar-background);

    > .mo-menu {
        height: $menuBar-size;
        color: var(--activityBar-inactiveForeground);
        background: inherit;
    }
}

// only the tabs in editor should have hover styles
.mo-editor__group {
    .mo-tab__item:hover {
        background-color: var(--tab-hoverBackground);
    }
}

.mo-tab__item__status,
.mo-tab__item__op {
    color: inherit;

    &:hover {
        color: inherit;
    }
}

.mo-editor__group-breadcrumb {
    color: var(--breadcrumb-foreground);

    .mo-breadcrumb__item:hover {
        color: var(--breadcrumb-focusForeground);
        opacity: 1;
    }
}

.mo-notification__item:focus {
    background: var(--list-hoverBackground);
    outline: 1px solid var(--list-focusOutline);
}

.mo-statusBar__item > a:empty {
    padding: 0;
}

// ===================== Auxiliary Bar ===========================
.mo-auxiliaryBar__tabs {
    .mo-auxiliaryBar__tabs__tab {
        font-size: 12px;
        padding-top: 5px;
        padding-bottom: 5px;

        &:hover {
            color: var(--taier-rightBar-foreground);
            background-color: var(--taier-rightBar-background);
        }

        &--active {
            color: var(--taier-rightBar-foreground);
            background-color: var(--taier-rightBar-background);
            position: relative;

            &::after {
                position: absolute;
                top: 0;
                right: 0;
                bottom: 0;
                width: 2px;
                height: 100%;
                background-color: #3f87ff;
                content: "";
            }
        }
    }
}

.mo-auxiliaryBar__container {
    background: var(--sideBar-background);
}
