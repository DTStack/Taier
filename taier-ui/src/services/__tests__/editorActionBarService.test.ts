import molecule from '@dtinsight/molecule';
import EditorActionBarService from '../editorActionBarService';
import { taskRenderService } from '..';

jest.mock(
    '@/services/executeService',
    () =>
        class {
            cbStore: any = {};
            onStartRun = (cb: any) => {
                this.cbStore.onStartRun = cb;
            };
            onEndRun = (cb: any) => {
                this.cbStore.onEndRun = cb;
            };
            onStopTab = (cb: any) => {
                this.cbStore.onStopTab = cb;
            };
            emit = (name: any, ...args: any[]) => {
                this.cbStore[name](...args);
            };
        }
);

jest.mock('@/services', () => ({
    taskRenderService: { renderEditorActions: jest.fn() },
}));

describe('The editor actionBar service', () => {
    beforeEach(() => {
        (molecule.editor.getState as jest.Mock).mockReset();
        (molecule.editor.getDefaultActions as jest.Mock).mockReset();
        (taskRenderService.renderEditorActions as jest.Mock).mockReset();
        (molecule.editor.updateActions as jest.Mock).mockReset();
    });

    it('Should init with empty set', () => {
        const service = new EditorActionBarService();
        const state = service.getState();
        expect(state.runningTab).toBeInstanceOf(Set);
        expect(state.runningTab.size).toBe(0);
    });

    it("Should support to update editor's actions by current tab", () => {
        (molecule.editor.getState as jest.Mock)
            .mockImplementationOnce(() => ({
                current: {
                    id: '1',
                    tab: {
                        data: {},
                    },
                },
            }))
            .mockImplementation(() => ({
                current: {
                    id: '1',
                },
            }));
        (taskRenderService.renderEditorActions as jest.Mock).mockImplementation(() => ['test']);
        (molecule.editor.getDefaultActions as jest.Mock).mockImplementation(() => ['actions']);

        const service = new EditorActionBarService();
        service.performSyncTaskActions();

        expect(molecule.editor.updateGroup).toBeCalledWith('1', { actions: ['test', 'actions'] });

        service.performSyncTaskActions();
        expect(molecule.editor.updateGroup).toBeCalledWith('1', { actions: ['actions'] });

        expect(molecule.editor.updateActions).not.toBeCalled();
    });

    it('Should update running task when current tab in running', () => {
        (molecule.editor.getState as jest.Mock).mockImplementation(() => ({
            current: {
                tab: {
                    data: {
                        id: 1,
                    },
                },
            },
        }));
        (taskRenderService.renderEditorActions as jest.Mock).mockImplementation(() => ['test']);
        (molecule.editor.getDefaultActions as jest.Mock).mockImplementation(() => ['actions']);

        const service = new EditorActionBarService();
        service.setState({
            runningTab: new Set([1]),
        });
        service.performSyncTaskActions();
        expect(molecule.editor.updateActions).toBeCalled();
    });

    it('Should listen execute service', () => {
        (molecule.editor.getState as jest.Mock).mockImplementation(() => ({
            current: {
                activeTab: '1',
            },
        }));
        const service = new EditorActionBarService();
        expect(service.getState().runningTab.size).toBe(0);

        // @ts-ignore
        service.executeService.emit('onStartRun', 1);

        expect(service.getState().runningTab.has(1)).toBeTruthy();
        // Change play action into running and enable stop action
        expect((molecule.editor.updateActions as jest.Mock).mock.calls[0][0]).toMatchSnapshot();

        // @ts-ignore
        service.executeService.emit('onEndRun', 1);
        expect(service.getState().runningTab.size).toBe(0);
        // Reset play action and disable stop action
        expect((molecule.editor.updateActions as jest.Mock).mock.calls[1][0]).toMatchSnapshot();

        // @ts-ignore
        service.executeService.emit('onStartRun', 1);
        // @ts-ignore
        service.executeService.emit('onStopTab', 1);
        // onStopTab is same with onEndRun
        expect(service.getState().runningTab.size).toBe(0);
        expect((molecule.editor.updateActions as jest.Mock).mock.calls[3][0]).toMatchSnapshot();
    });
});
