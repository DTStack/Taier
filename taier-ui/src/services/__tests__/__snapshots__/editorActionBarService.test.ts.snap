// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The editor actionBar service Should listen execute service 1`] = `
Array [
  Object {
    "disabled": true,
    "icon": "loading~spin",
    "id": "task.run",
    "name": "Running",
    "title": "运行中",
  },
  Object {
    "disabled": false,
    "id": "task.stop",
  },
]
`;

exports[`The editor actionBar service Should listen execute service 2`] = `
Array [
  Object {
    "disabled": false,
    "icon": "play",
    "id": "task.run",
    "name": "Run Task",
    "place": "outer",
    "title": "运行",
  },
  Object {
    "disabled": true,
    "icon": "debug-pause",
    "id": "task.stop",
    "name": "Stop Task",
    "place": "outer",
    "title": "停止运行",
  },
]
`;

exports[`The editor actionBar service Should listen execute service 3`] = `
Array [
  Object {
    "disabled": false,
    "icon": "play",
    "id": "task.run",
    "name": "Run Task",
    "place": "outer",
    "title": "运行",
  },
  Object {
    "disabled": true,
    "icon": "debug-pause",
    "id": "task.stop",
    "name": "Stop Task",
    "place": "outer",
    "title": "停止运行",
  },
]
`;
